### DISCUSSION TITLE
Bind to Unix Socket ?

### DISCUSSION THREAD

**Discussion by mikehu404** [umputun/remark42#1602] [Ideas]
Hello,
I wonder if remark42 server could bind to unix socket,
I can't find it anywhere in remark42's doc.
Unix socket is faster than tcp, as discussed [here](https://stackoverflow.com/a/29436429)
It would be great to have this feature.
Especially if you want to put remark42 behind a reverse proxy like nginx or haproxy.

**Comment by umputun** [comment]
pls use discussions unless you report an actual bug

moved

**Comment by umputun** [comment]
If the purpose of this idea is to improve performance, it should be tested first using an appropriate set of benchmarks. In my opinion, I don't believe this change would be noticeable in comparison to all the other aspects involved in request handling and unix sockets are unlikely to have a significant impact on performance.

However, the change is trivial and you can try to benchmark it. If my prediction is wrong and we see a material improvement we can consider adding support of unix sockets

**Reply by paskal** [reply]
My vote is against implementing it. Nobody cares about improving connection time from 6 microseconds to 2 microseconds when you need to send a bunch of bytes over the internet to the user so that they can render it: it will take them at 50ms minimum to get the comments (best-case scenario, but already 10 000 times bigger number than both 2 and 6 microseconds) and 30-200ms to render it based on how fast their device is.
